src/models/shared/shared-model-manager.ts

```diff
@@ -118,10 +117,10 @@ export interface ISharedModelManager {
 
   /**
    * Get the tiles that link to this shared model
-   * 
-   * @param sharedModel 
+   *
+   * @param sharedModel
    */
-  getSharedModelTiles(sharedModel?: SharedModelType): ITileModel[];
+  getSharedModelTiles(sharedModel?: SharedModelType): IAnyStateTreeNode[];
```

Comment:
I guess the removal of the loop is more beneficial than the typing.
In most cases we "cast" the return value of this function anyway.
And this is now symmetrical to `getTileSharedModels` above.

Looking at the shared folder I also see that `shared-model-document-manager.ts` depends on both ITileModel and DocumentContentModelType. That could be moved to the `document` folder so then I think the shared folder would not have any outside references.